File: oa_questions/leetcode/maxConsecutiveOnes.java
/**
 * Maintain current count and maximum count so far.
 *
 * Time Complexity: O(N)
 *
 * Space Complexity: O(1)
 *
 * N = Length of the input array
 */
 class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        if (nums == null) {
            throw new IllegalArgumentException("Input array is null");
        }

        int curCount = 0;
        int maxCount = 0;

        for (int n : nums) {
            if (n == 1) {
                curCount++;
                maxCount = Math.max(maxCount, curCount);
            } else {
                curCount = 0;
            }
        }

        return maxCount;
    }
}

File: oa_questions/leetcode/maxConsecutiveOnes.js
var findMaxConsecutiveOnes = function (nums) {
  let max = 0,
    curr = 0;
  for (let k of nums) {
    max = Math.max(max, (curr += k));
    if (!k) curr = 0;
  }
  return max;
};

var findMaxConsecutiveOnes = function (nums) {
  return nums
    .join("")
    .split("0")
    .reduce((max, ones) => Math.max(max, ones.length), 0);
};

var findMaxConsecutiveOnes = function (nums) {
  return Math.max(
    ...nums
      .join("")
      .split("0")
      .map((ones) => ones.length)
  );
};

var findMaxConsecutiveOnes = function (nums) {
  return Math.max(
    ...(nums.join("").match(/1+/g) || [""]).map((ones) => ones.length)
  );
};
